### DISCUSSION TITLE
Transparency guidance

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/RiverscapesXML#112] [RAVE: Symbology/BusinessLogic] [answered]
We have a lot more control in QGIS with *.qml over transparency settings then we do in Arc with *.lyr files that only include path information and symbology. We have a `transparency` tag in the business logic because in ArcRAVE you needed it (i.e. controlled in display tab, but not stored with *.lyr):
![image](https://user-images.githubusercontent.com/9723905/122263168-5bc3b900-ce93-11eb-872a-32565eafa1c9.png)

However individual authors of symbology are inconsistent in QGIS whether or not they use this business logic setting setting and always leave layer transparent. So when someone saves this in both their *.qml and in the business logic, what happens:

`Node label="Likelihood of Being a Valley Bottom Raster" xpath="Raster[@id='VBET_EVIDENCE']" type="raster" symbology="vbetLikelihood" transparency="40" />`

Does it apply both and make it really, really transparent? Or does it override one? More to the point, what should we be doing @MattReimer? It seems we should instruct people to only control it in the business logic?

**Comment by MattReimer** [comment, marked as answer]
Ok, this is probably a bigger topic than we all want it to be and the answer is "it varies"

But there are more complex cases:

Here's a vector where only purple and blue values are transparent
<img width="557" alt="Screen Shot 2021-06-16 at 10 44 43 AM" src="https://user-images.githubusercontent.com/1063391/122268020-6af53780-ce90-11eb-92ef-b4eb0f49524f.png">

Here's a case where shapes with an id that is divisible by 2 have solid lines, triangle fill but the space between is transparent
<img width="508" alt="Screen Shot 2021-06-16 at 10 48 11 AM" src="https://user-images.githubusercontent.com/1063391/122268075-7e080780-ce90-11eb-92f0-ff50f76bbde9.png">

My rules are as follows:

If it's simple and you just want the whole layer to be transparent, use the businesslogic field and don't make any transparency in your QML file at all

If it's something more complex, leave the businesslogic field alone and bake it into the QML

**Reply by joewheaton** [reply]
Thanks @MattReimer. I think that works for now. Lets all the @Riverscapes/riverscape-project-curators keep an eye on this and share thoughts here. For now, I'm going to mark this as the answer.
